test/index_spec.js

```diff
@@ -218,5 +218,65 @@ describe('textPosition', () => {
       let range = toRange(fixture.el)
       assert.isTrue(range.collapsed)
     })
+
+    it('returns a range selecting the first text of the root element', () => {
+      let root = fixture.el
+      let expected = 'Pellentesque'
+      let start = 0
+      let end = expected.length
+      let range = toRange(root, {start, end})
+      let text = range.toString()
+      assert.equal(text, expected)
+    })
+
+    it('returns a range selecting the last text of the root element', () => {
+      let root = fixture.el
+      let expected = 'erat.'
+      let end = root.textContent.length
+      let start = end - 5
+      let range = toRange(root, {start, end})
+      let text = range.toString()
+      assert.equal(text, expected)
+    })
+
+    it('returns an empty range selecting the end of the root element', () => {
+      let root = fixture.el
+      let end = root.textContent.length
+      let range = toRange(root, {start: end, end})
+      let text = range.toString()
+      assert.equal(text, '')
+    })
+
+    it('throws an error if the start offset is out of range', () => {
+      let root = fixture.el
+      assert.throws(() => {
+        let start = 10000
+        let end = start + 1
+        toRange(root, {start, end})
+      }, 'Start offset of position selector is out of range')
+    })
+
+    it('throws an error if the end offset is out of range', () => {
+      let root = fixture.el
+      assert.throws(() => {
+        let start = 0
+        let end = 10000
+        toRange(root, {start, end})
+      }, 'End offset of position selector is out of range')
+    })
+
+    it('handles an empty root element', () => {
+      let root = document.createElement('div');
+      assert.throws(() => {
+        toRange(root, {start:0, end: 0})
```

Comment:
Seems like it shouldn't throw it this case, since `[0, 0)` is a valid, empty range.

Reply:
I might be convinced it should, but I've gotta think about it a little bit.

Reply:
My thinking here is that there is an implicit invariant in all other cases that the returned range starts and ends in a text node. If there are no text nodes, that's not possible.

The other motivation was simply convenience as it avoids needing to special-case this in the implementation.